File: ngDesk/ngDesk-Rest/src/main/java/com/ngdesk/modules/detail/layouts/DCELayout.java
package com.ngdesk.modules.detail.layouts;

import java.util.Date;
import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties("_class")
public class DCELayout {

	@JsonProperty("LAYOUT_ID")
	private String layoutId;

	@JsonProperty("NAME")
	@NotNull(message = "LAYOUT_NAME_NOT_NULL")
	@Size(min = 1, message = "LAYOUT_NAME_NOT_EMPTY")
	private String name;

	@JsonProperty("DESCRIPTION")
	@NotNull(message = "DESCRIPTION_NOT_NULL")
	private String description;

	@JsonProperty("PANELS")
	@Valid
	private List<Panel> panels;

	@JsonProperty("TITLE_BAR")
	@Valid
	private List<TitleBarGrid> titleBar;

	@JsonProperty("PREDEFINED_TEMPLATE")
	@Valid
	private List<PreDefinedTemplate> preDefinedTemplate;

	@JsonProperty("CUSTOM_LAYOUT")
	private String customLayout;

	@JsonProperty("ROLE")
	@NotNull(message = "ROLE_NOT_NULL")
	private String role;

	@JsonProperty("DATE_CREATED")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
	private Date dateCreated;

	@JsonProperty("DATE_UPDATED")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
	private Date dateUpdated;

	@JsonProperty("LAST_UPDATED_BY")
	private String lastUpdatedBy;

	@JsonProperty("CREATED_BY")
	private String createdBy;

	@JsonProperty("LAYOUT_STYLE")
	@Pattern(regexp = "^(fill|standard|outline)$", message = "INVALID_LAYOUT_STYLE")
	private String layoutStyle;

	public DCELayout() {
	}

	public DCELayout(String layoutId,
			@NotNull(message = "LAYOUT_NAME_NOT_NULL") @Size(min = 1, message = "LAYOUT_NAME_NOT_EMPTY") String name,
			@NotNull(message = "DESCRIPTION_NOT_NULL") String description, @Valid List<Panel> panels,
			@Valid List<TitleBarGrid> titleBar, @Valid List<PreDefinedTemplate> preDefinedTemplate, String customLayout,
			@NotNull(message = "ROLE_NOT_NULL") String role, Date dateCreated, Date dateUpdated, String lastUpdatedBy,
			String createdBy,
			@Pattern(regexp = "^(fill|standard|outline)$", message = "INVALID_LAYOUT_STYLE") String layoutStyle) {
		super();
		this.layoutId = layoutId;
		this.name = name;
		this.description = description;
		this.panels = panels;
		this.titleBar = titleBar;
		this.preDefinedTemplate = preDefinedTemplate;
		this.customLayout = customLayout;
		this.role = role;
		this.dateCreated = dateCreated;
		this.dateUpdated = dateUpdated;
		this.lastUpdatedBy = lastUpdatedBy;
		this.createdBy = createdBy;
		this.layoutStyle = layoutStyle;
	}

	public String getLayoutId() {
		return layoutId;
	}

	public void setLayoutId(String layoutId) {
		this.layoutId = layoutId;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public List<Panel> getPanels() {
		return panels;
	}

	public void setPanels(List<Panel> panels) {
		this.panels = panels;
	}

	public List<TitleBarGrid> getTitleBar() {
		return titleBar;
	}

	public void setTitleBar(List<TitleBarGrid> titleBar) {
		this.titleBar = titleBar;
	}

	public List<PreDefinedTemplate> getPreDefinedTemplate() {
		return preDefinedTemplate;
	}

	public void setPreDefinedTemplate(List<PreDefinedTemplate> preDefinedTemplate) {
		this.preDefinedTemplate = preDefinedTemplate;
	}

	public String getCustomLayout() {
		return customLayout;
	}

	public void setCustomLayout(String customLayout) {
		this.customLayout = customLayout;
	}

	public String getRole() {
		return role;
	}

	public void setRole(String role) {
		this.role = role;
	}

	public Date getDateCreated() {
		return dateCreated;
	}

	public void setDateCreated(Date dateCreated) {
		this.dateCreated = dateCreated;
	}

	public Date getDateUpdated() {
		return dateUpdated;
	}

	public void setDateUpdated(Date dateUpdated) {
		this.dateUpdated = dateUpdated;
	}

	public String getLastUpdatedBy() {
		return lastUpdatedBy;
	}

	public void setLastUpdatedBy(String lastUpdatedBy) {
		this.lastUpdatedBy = lastUpdatedBy;
	}

	public String getCreatedBy() {
		return createdBy;
	}

	public void setCreatedBy(String createdBy) {
		this.createdBy = createdBy;
	}

	public String getLayoutStyle() {
		return layoutStyle;
	}

	public void setLayoutStyle(String layoutStyle) {
		this.layoutStyle = layoutStyle;
	}

}

File: ngDesk-UI/src/app/models/dce-layout.ts
import { DceColumn } from './dce-column';

export class DceLayout {
	constructor(
		private NAME: string,
		private DESCRIPTION: string,
		private ROLE: string,
		private DATE_CREATED?: Date,
		private DATE_UPDATED?: Date,
		private LAST_UPDATED_BY?: string,
		private CREATED_BY?: string,
		private COLUMNS?: DceColumn[],
		private CUSTOM_LAYOUT?: string,
		private LAYOUT_ID?: string,
		private GRIDS?: any[][],
		private PREDEFINED_TEMPLATE?: any[]
	) {}

	get predefinedTemplate() {
		return this.PREDEFINED_TEMPLATE;
	}

	set predefinedTemplate(predefinedTemplate) {
		this.PREDEFINED_TEMPLATE = predefinedTemplate;
	}
	get grids() {
		return this.GRIDS;
	}

	set grids(grids) {
		this.GRIDS = grids;
	}

	get customLayout() {
		return this.CUSTOM_LAYOUT;
	}

	set customLayout(customLayout) {
		this.CUSTOM_LAYOUT = customLayout;
	}

	get layoutId() {
		return this.LAYOUT_ID;
	}

	set layoutId(value) {
		this.LAYOUT_ID = value;
	}

	get name() {
		return this.NAME;
	}

	set name(value) {
		this.NAME = value;
	}

	get description() {
		return this.DESCRIPTION;
	}

	set description(value) {
		this.DESCRIPTION = value;
	}

	get columns() {
		return this.COLUMNS;
	}

	set columns(value) {
		this.COLUMNS = value;
	}

	get role() {
		return this.ROLE;
	}

	set role(value) {
		this.ROLE = value;
	}

	get dateCreated() {
		return this.DATE_CREATED;
	}

	set dateCreated(value) {
		this.DATE_CREATED = value;
	}

	get dateUpdated() {
		return this.DATE_UPDATED;
	}

	set dateUpdated(value) {
		this.DATE_UPDATED = value;
	}

	get lastUpdatedBy() {
		return this.LAST_UPDATED_BY;
	}

	set lastUpdatedBy(value) {
		this.LAST_UPDATED_BY = value;
	}

	get createdBy() {
		return this.CREATED_BY;
	}

	set createdBy(value) {
		this.CREATED_BY = value;
	}
}
